<a href="https://colab.research.google.com/github/denise25maciel/analiseDadosPython/blob/main/vd3_police.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Projeto 3: avaliação de dados policiais

# Etapa1: importanto arquivos e bibliotecas



In [1]:
import pandas as pd

In [3]:
df= pd.read_csv('/content/vd3_dadospolicia.csv')

# Etapa2: conhecendo os dados

In [5]:
df.head(5)

stop_date stop_time  ...  stop_duration drugs_related_stop
0   1/2/2005      1:55  ...       0-15 Min              False
1  1/18/2005      8:15  ...       0-15 Min              False
2  1/23/2005     23:15  ...       0-15 Min              False
3  2/20/2005     17:15  ...      16-30 Min              False
4  3/14/2005     10:00  ...       0-15 Min              False

[5 rows x 15 columns]

In [6]:
df.columns

Index(['stop_date', 'stop_time', 'country_name', 'driver_gender',
       'driver_age_raw', 'driver_age', 'driver_race', 'violation_raw',
       'violation', 'search_conducted', 'search_type', 'stop_outcome',
       'is_arrested', 'stop_duration', 'drugs_related_stop'],
      dtype='object')

In [8]:
df.shape

(65535, 15)

In [4]:
df.isnull().sum()

stop_date                 0
stop_time                 0
country_name          65535
driver_gender          4061
driver_age_raw         4054
driver_age             4307
driver_race            4060
violation_raw          4060
violation              4060
search_conducted          0
search_type           63056
stop_outcome           4060
is_arrested            4060
stop_duration          4060
drugs_related_stop        0
dtype: int64

Nota-se que todos os valores da coluna 'country_name' são nulos. Dessa forma a coluna será removida

# Etapa 3: respondendo questões sobre os dados

# Q1: como remover a coluna 'country_name'?

In [10]:
df.drop(columns ='country_name', inplace =True )

In [11]:
df.columns

Index(['stop_date', 'stop_time', 'driver_gender', 'driver_age_raw',
       'driver_age', 'driver_race', 'violation_raw', 'violation',
       'search_conducted', 'search_type', 'stop_outcome', 'is_arrested',
       'stop_duration', 'drugs_related_stop'],
      dtype='object')

# Q2:Com que frequencia, homens e mulheres realizam violação por conta da velocidade (Speeding)? Coluna violation.

In [17]:
df.columns


Index(['stop_date', 'stop_time', 'driver_gender', 'driver_age_raw',
       'driver_age', 'driver_race', 'violation_raw', 'violation',
       'search_conducted', 'search_type', 'stop_outcome', 'is_arrested',
       'stop_duration', 'drugs_related_stop'],
      dtype='object')

In [22]:
df[df.violation =='Speeding'].driver_gender.value_counts()

M    25517
F    11686
Name: driver_gender, dtype: int64

# Q3: o gênero afeta quem é revisado durante uma parada?

In [24]:
df.columns

Index(['stop_date', 'stop_time', 'driver_gender', 'driver_age_raw',
       'driver_age', 'driver_race', 'violation_raw', 'violation',
       'search_conducted', 'search_type', 'stop_outcome', 'is_arrested',
       'stop_duration', 'drugs_related_stop'],
      dtype='object')

In [39]:
df[df.search_conducted ==True].driver_gender.value_counts()

M    2113
F     366
Name: driver_gender, dtype: int64

In [43]:
#O mesmo resultado pode ser utilizado através do group_by
df.groupby('driver_gender').search_conducted.sum()

Logo, o número de homens revisados é superior

In [45]:
df.search_conducted.value_counts()

False    63056
True      2479
Name: search_conducted, dtype: int64

Dentre 2476 pessoas conduzidas, 2113 são homens e 366 são mulheres

# Q4 Qual o tempo médio de duração das paradas?

A coluna stop_duration está no formato de string, logo será necessário realizar conversão dos dados

In [46]:
df.columns

Index(['stop_date', 'stop_time', 'driver_gender', 'driver_age_raw',
       'driver_age', 'driver_race', 'violation_raw', 'violation',
       'search_conducted', 'search_type', 'stop_outcome', 'is_arrested',
       'stop_duration', 'drugs_related_stop'],
      dtype='object')

In [49]:
df['stop_duration'].value_counts()

0-15 Min     47379
16-30 Min    11448
30+ Min       2647
2                1
Name: stop_duration, dtype: int64

Como os dados foram fornecidos em intervalos de tempo, optou-se por utilizar o tempo médio

In [53]:
df['stop_duration'] = df['stop_duration'].map({
    '0-15 Min':7.5,
    '16-30 Min':24,
    '30+ Min':45
})

Nessa coluna possuímos valores nulos, substituíremos os mesmos pela média DOS VALORES PREENCHIDOS

In [60]:
df['stop_duration'].isnull().sum()

4061

In [61]:
df['stop_duration'].value_counts()

7.5     47379
24.0    11448
45.0     2647
Name: stop_duration, dtype: int64

In [63]:
df['stop_duration'].fillna(df['stop_duration'].mean(), inplace=True)


In [64]:
df['stop_duration'].isnull().sum()

0

In [65]:
df['stop_duration'].mean()

12.187420698182255

# Q5: compare a distribuição de idade para cada visualização

In [69]:
df.groupby('violation').driver_age.describe()


count       mean        std  ...   50%   75%   max
violation                                           ...                  
Equipment             6507.0  31.682957  11.380671  ...  28.0  39.0  81.0
Moving violation     11876.0  36.736443  13.258350  ...  35.0  47.0  86.0
Other                 3477.0  40.362381  12.754423  ...  41.0  50.0  86.0
Registration/plates   2240.0  32.656696  11.150780  ...  30.0  40.0  74.0
Seat belt                3.0  30.333333  10.214369  ...  26.0  34.0  42.0
Speeding             37120.0  33.262581  12.615781  ...  30.0  42.0  88.0

[6 rows x 8 columns]

In [71]:
df.groupby(['violation','driver_age']).size()


violation  driver_age
Equipment  16.0            4
           17.0           32
           18.0          130
           19.0          281
           20.0          348
                        ... 
Speeding   82.0            7
           83.0            1
           84.0            3
           86.0            3
           88.0            2
Length: 333, dtype: int64